esi/lib/networks.py

```diff
@@ -85,22 +98,19 @@ def get_networks_from_port(connection, port, networks_dict={}, floating_ips_dict
         parent_network = connection.network.get_network(network=port.network_id)
 
     if port.trunk_details:
-        with concurrent.futures.ThreadPoolExecutor() as executor:
```

Comment:
Oh, simply because it should no longer be necessary. The port dictionary should contain all the ports the user can see, so the API query should never be made. The `if/else` is still there mostly as a just-in-case thing - similar to what the code does with networks - but uncluttering the code feels like a win to me.